src/index.ts

```diff
@@ -0,0 +1,53 @@
+export enum lockClasses {
+    simple = 'is-blocked',
+    ios = 'is-blocked-touch'
+}
+
+export class PageLocker {
+    private _initialScrollPosition = 0;
+
+    private _retainers: { [id: string]: boolean } = {};
+
+    private _isLocked = false;
+
+    private _target = document.documentElement;
+
+    private _isIOS: boolean;
+
+    private _lockedClass: string;
+
+    constructor() {
+        this._isIOS = !!navigator.platform.match(/iPhone|iPod|iPad/);
+        this._lockedClass = this._isIOS ? lockClasses.ios : lockClasses.simple;
+    }
+
+    public lock = (id: string): void => {
+        this._retainers[id] = true;
+        const { _target, _isIOS, _lockedClass } = this;
```

Comment:
Just thinking about that line of code. Do we really need to use destructuring assignment here? We've already had all we need in "this" keyword and can easily use it. 

Apart from that, we see the mix of using "this" keyword and variables from the destructive assignment. e.g 
` if (!this._isLocked) {
            if (_isIOS)) {}
}`

What thoughts on it, guys @Ofigelov  @seregap  @bububo? Is it just a code style question or we have some guidelines on it?

Reply:
My minds about it, that if we use something twice ore more in the current block, we should destruct it for shorter and more readable code. And yes, I have to add `_initialScrollPosition` in destructure =)
Maybe `_isLocked` too, but I don't sure would destructed element influence on his parent or not.